roles/kube_proxy_and_dns/files/kube-proxy-and-dns.yaml

```diff
@@ -81,6 +81,7 @@ spec:
           oc config --config=/tmp/kubeconfig set-context "$( oc config --config=/tmp/kubeconfig current-context )" --user=sa
           # Launch the kube-proxy and DNS process
           exec openshift start network --disable=plugins --enable=proxy,dns --config=/etc/origin/node/node-config.yaml --kubeconfig=/tmp/kubeconfig --loglevel=${DEBUG_LOGLEVEL:-2}
+          dbus-send --system --dest=uk.org.thekelleys.dnsmasq /uk/org/thekelleys/dnsmasq uk.org.thekelleys.SetDomainServers array:string:
```

Comment:
@Miciah Do you know if this command will actually execute when the previous exec exits abnormally?

Reply:
Now I also wonder if `dbus-send` exiting 0 after the previous `exec` exits nonzero masks the `exec` error...

Reply:
Same concern- won't that prevent nonzero return of `openshift start network` from ever being observed?

Reply:
If we need to preserve the return value, we can easily do so:
```bash
          ret=0
          openshift start network --disable=plugins --enable=proxy,dns --config=/etc/origin/node/node-config.yaml --kubeconfig=/tmp/kubeconfig --loglevel=${DEBUG_LOGLEVEL:-2} || ret=$?
          if [[ $ret -ne 0 ]]; then
            dbus-send --system --dest=uk.org.thekelleys.dnsmasq /uk/org/thekelleys/dnsmasq uk.org.thekelleys.SetDomainServers array:string:
          fi
          exit $ret
```

Reply:
I would say we need to, otherwise how could we (or the kubelet, etc) ever detect a failure of the primary process this container represents?

Reply:
Thanks Miciah.

Reply:
Sorry, @ironcladlou, I only saw your latest comment after I posted my previous comment.  Yeah, that makes sense.

I think unconditional cleanup also makes sense—it doesn't look like OpenShift cleans up the configuration even on a normal exit, and even if it did, it shouldn't hurt anything to clear it twice.